sqlair.go

```diff
@@ -156,6 +176,9 @@ func (iter *Iterator) Close() error {
 func (q *Query) One(outputArgs ...any) error {
 	iter := q.Iter()
 	if !iter.Next() {
+		if iter.err != nil {
```

Comment:
We should be able to do this without breaking the Iter abstraction and depending on details of its implementation here, or calling Close twice:
```
err := ErrNoRows
iter := q.Iter()
if iter.Next() {
    iter.Decode(outputArgs...)
    err = nil
}
if cerr := iter.Close(); cerr != nil {
        return cerr
}
return err
```

Reply:
Changed.